code/modules/power/apc/apc.dm

```diff
@@ -0,0 +1,1064 @@
+#define CHANNEL_SETTING_OFF 0
+/// APC power channel Setting Off, if set while apc is "on" set apc to "off" otherwise set to "auto-off"
+#define CHANNEL_SETTING_AUTO_OFF 1
+/// APC power channel setting on,
+#define CHANNEL_SETTING_ON 2   //on
+// APC user setting,
+#define CHANNEL_SETTING_AUTO_ON  3 //auto
+
+// the Area Power Controller (APC), formerly Power Distribution Unit (PDU)
+// one per area, needs wire conection to power network through a terminal
+
+// controls power to devices in that area
+// may be opened to change power cell
+// three different channels (lighting/equipment/environ) - may each be set to on, off, or auto
+
+/obj/machinery/power/apc
+	name = "area power controller"
+	desc = "A control terminal for the area electrical systems."
+	icon_state = "apc0"
+	max_integrity = 200
+	integrity_failure = 50
+	resistance_flags = FIRE_PROOF
+	req_access = list(ACCESS_ENGINE_EQUIP)
+	siemens_strength = 1
+	damage_deflection = 10
+	move_resist = INFINITY
+
+	// set so that APCs aren't found as powernet nodes //Hackish, Horrible, was like this before I changed it :(
+	powernet = 0
+
+	/*** APC Area/Powernet vars ***/
+	/// the area that this APC is in
+	var/area/apc_area
+	/// the set string name of the area, used in naming the APC
+	var/areastring = null
+	/// The power terminal connected to this APC
+	var/obj/machinery/power/terminal/terminal = null
+	/// The status of the terminals powernet that this APC is connected to: not connected, no power, or recieving power
+	var/main_status = APC_EXTERNAL_POWER_NOTCONNECTED
+
+	/// amount of power used in the last cycle for lighting channel
+	var/last_used_lighting = 0
+	/// amount of power used in the last cycle for equipment channel
+	var/last_used_equipment = 0
+	/// amount of power used in the last cycle for environment channel
+	var/last_used_environment = 0
+	/// amount of power used in the last cycle in total
+	var/last_used_total = 0
+
+	/*** APC Cell Vars ***/
+	/// the cell type stored in this APC
+	var/obj/item/stock_parts/cell/cell
+	/// the percentage charge the internal battery will start with
+	var/start_charge = 90
+	/// Base cell has 2500 capacity. Enter the path of a different cell you want to use. cell determines charge rates, max capacity, ect. These can also be changed with other APC vars, but isn't recommended to minimize the risk of accidental usage of dirty editted APCs
+	var/cell_type = 2500
+
+	/*** APC Status Vars ***/
+	/// The wire panel associated with this APC
+	var/datum/wires/apc/wires = null
+	/// Can the APC recieve/transmit power? Determined by the condition of the 2 Main Power Wires
+	var/shorted = FALSE
+	/// Is the APC on and transmitting power to enabled breakers? Think of this var as the master breaker for the APC
+	var/operating = TRUE
+	/// The current charging mode of the APC: not charging, charging, fully charged
+	var/charging = APC_NOT_CHARGING
+	var/longtermpower = 10
```

Comment:
A comment for this var would be nice too ❤️ 

Also, probably should be snake case.

Reply:
comment added, will try and get snake case done as well